sunpy/coordinates/wcs_utils.py

```diff
@@ -52,6 +59,29 @@ def solar_wcs_frame_mapping(wcs):
         return Heliocentric(obstime=dateobs, observer=observer)
 
 
+def _set_wcs_aux_obs_coord(wcs, obs_coord):
+    """
+    Set (in-place) observer coordinate information on a WCS.
+
+    Parameters
+    ----------
+    wcs : astropy.wcs.WCS
+    obs_coord : astropy.coordinates.SkyCoord
+    """
+    # Sometimes obs_coord can be a frame (with data); convert to a SkyCoord to
+    # ensure the .frame attribute is present
+    obs_coord = SkyCoord(obs_coord)
```

Comment:
I suggest you do this the other way around:
```
if not isinstance(obs_coord, CoordinateFrame):
    obs_coord = obs_coord.frame
```
the SkyCoord constructor is very heavy we don't want to call it if we don't have to.